### PR TITLE
ref(api4/cloud): remove direct store call

### PR DESCRIPTION
**Pull request by majidsajadi**
#### Summary
Remove direct Store calls from the API layer for the file `api4/cloud.go`

#### Ticket Link
None. but it almost like #15750 

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by majidsajadi** [comment]
@jespino

File: api4/cloud.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func getSubscriptionStats(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	count, err := c.App.Srv().Store.User().Count(model.UserCountOptions{})
+	count, err := c.App.Srv().GetTotalUsersCount()
```
Comment (jespino) [inline api4/cloud.go:86]: You are calling `GetTotalUsersCount` in the `Srv()` and you should be calling this in the `c.App`.

Comment (majidsajadi) [reply api4/cloud.go]: My bad. fixed. @jespino

**Review by wiggin77 — DISMISSED** [review]
LGTM

**Comment by wiggin77** [comment]
You need to run `make app-layers` and commit the results.

**Comment by majidsajadi** [comment]
@wiggin77 Done.

File: app/user.go [hunk 2]
```diff
@@ -2479,3 +2479,7 @@ func (a *App) UpdateThreadReadForUser(userID, teamID, threadID string, timestamp
 	a.Publish(message)
 	return thread, nil
 }
+
+func (a *App) GetTotalUsersCount() (int64, error) {
+	return c.App.Srv().Store.User().Count(model.UserCountOptions{})
```
Comment (jespino) [inline app/user.go:2484]: There is no `c` in this context, is just `a.Srv()...`

File: api4/cloud.go [hunk 3]
```diff
@@ -83,7 +83,7 @@ func getSubscriptionStats(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	count, err := c.App.Srv().Store.User().Count(model.UserCountOptions{})
+	count, err := c.App.GetTotalUsersCount()
 	if err != nil {
 		c.Err = model.NewAppError("Api4.getSubscriptionStats", "app.user.get_total_users_count.app_error", nil, err.Error(), http.StatusInternalServerError)
```
Comment (jespino) [inline api4/cloud.go:88]: I think I would prefer to create this `AppError` from the new created app function, and here just use the `c.Err = err`.

Comment (majidsajadi) [reply api4/cloud.go:88]: @jespino  if i create this error in the app layer `where` param of `NewAppError`, should be "GetTotalUsersCount". is it okay? 
I'm new to this project and I don't exactly know how does error handling work. 

so I change `GetTotalUsersCount` to: 

```go
func (a *App) GetTotalUsersCount() (int64, error) {
  count, err := a.Srv().Store.User().Count(model.UserCountOptions{})
  if err != nil {
    return nil, model.NewAppError("GetTotalUsersCount", "app.user.get_total_users_count.app_error", nil, err.Error(), http.StatusInternalServerError)
  }
}
```

and then I call that like this: 

```go
count, err := c.App.GetTotalUsersCount()
if err != nil {
    c.Err = err
}
```

let me know if its okay.

**Review by jespino — DISMISSED** [review]
Now looks good to me. Thanks @majidsajadi! :tada:

File: app/user.go [hunk 4]
```diff
@@ -2479,3 +2479,11 @@ func (a *App) UpdateThreadReadForUser(userID, teamID, threadID string, timestamp
 	a.Publish(message)
 	return thread, nil
 }
+
+func (a *App) GetTotalUsersCount() (int64, error) {
+	count, err := a.Srv().Store.User().Count(model.UserCountOptions{})
+	if err != nil {
+		return nil, model.NewAppError("GetTotalUsersCount", "app.user.get_total_users_count.app_error", nil, err.Error(), http.StatusInternalServerError)
```
Comment (jespino) [inline app/user.go:2486]:
```suggestion
		return 0, model.NewAppError("GetTotalUsersCount", "app.user.get_total_users_count.app_error", nil, err.Error(), http.StatusInternalServerError)
```

**Comment by jespino** [comment]
@majidsajadi I fixed a tiny problem, you were returning `nil` as an `int64` so it was failing. now should be fine. :)

**Comment by isacikgoz** [comment]
Hi @majidsajadi, big thanks for your contribution! There hasn't been recent activity on this PR so I've marked it as inactive. Following our [inactive contribution process](https://developers.mattermost.com/contribute/getting-started/inactive-contributions/), the PR is now closed due to 30 days or more of inactivity.

However, if you're still interested working on the changes, let us know! Note that we [merged multiple repositories into one monorepo](https://github.com/mattermost/mattermost-server/issues/22420), which may require you to resubmit the changes. 

If you have any questions, don't hesitate to let us know, we're happy to help.

**Pull request closed without merging**